python/ray/data/read_api.py

```diff
@@ -322,6 +325,85 @@ def read_datasource(
+
+    The data to read from is specified via the ``uri``, ``database`` and ``collection``
+    of the MongoDB. The pipeline is executed against the specified collection. The
+    execution results are converted into Arrow format and then used to create a dataset.
+
+    You can check out more details here about these MongoDB concepts:
+    - URI: https://www.mongodb.com/docs/manual/reference/connection-string/
+    - Database and Collection: https://www.mongodb.com/docs/manual/core/databases-and-collections/
+    - Pipeline: https://www.mongodb.com/docs/manual/core/aggregation-pipeline/
+
+    To read the MongoDB in parallel, the execution of the pipeline is sharded, with a
+    Ray read task to handle a partition of the results. The number of partitions is
+    determined by ``parallelism`` which can be requested from this interface or
+    automatically chosen if unspecified. The sharding is automatically determined
+    in an attempt to evenly distribute the documents into the specified number of
+    partitions.
+
+    Examples:
+        >>> import ray
+        >>> from pymongoarrow.api import Schema
+        >>> ds = ray.data.read_mongo( # doctest: +SKIP
+        ...     uri="mongodb://username:password@mongodb0.example.com:27017/?authSource=admin", # noqa: E501
+        ...     database="my_db",
+        ...     collection="my_collection",
+        ...     pipeline=[{"$match": {"col2": {"$gte": 0, "$lt": 100}}}, {"$sort": "sort_field"}], # noqa: E501
+        ...     schema=Schema({"col1": pa.string(), "col2": pa.int64()}),
+        ...     parallelism=10,
+        ... )
+
+    Args:
+        uri: The URI of the source MongoDB where the dataset will be
+            read from. For the URI format, see details in
+            https://www.mongodb.com/docs/manual/reference/connection-string/.
+        database: The name of the database hosted in the MongoDB. This database
+            must exist otherwise ValueError will be raised.
+        collection: The name of the collection in the database. This collection
+            must exist otherwise ValueError will be raised.
+        pipeline: A MongoDB pipeline, which will be executed on the given collection
+            with results used to create Dataset. If None, the entire collection will
+            be read.
+        schema: The schema used to read the collection. If None, it'll be inferred from
+            the results of pipeline.
+        parallelism: The requested parallelism of the read. If -1, it will be
+            automatically chosen based on the available cluster resources and estimated
+            in-memory data size.
+        ray_remote_args: kwargs passed to ray.remote in the read tasks.
+        mong_args: kwargs passed to aggregate_arrow_all() in pymongoarrow in producing
+            Arrow-formatted results.
+
+    Returns:
+        Dataset holding Arrow records from the results of executing the pipeline on the
+        specified MongoDB collection.
```

Comment:
Dataset holding Arrow records from the specified MongoDB collection.

Reply:
I think we need to call out it's the execution results of pipeline that's gotten into dataset.